classes/submission/Collector.php

```diff
@@ -383,40 +419,79 @@ public function getQueryBuilder(): Builder
+        // Search phrase
+        if ($keywords->count()) {
             $likePattern = DB::raw("CONCAT('%', LOWER(?), '%')");
-            $words = explode(' ', $this->searchPhrase);
-            foreach ($words as $word) {
-                $q->where(function ($query) use ($word, $likePattern) {
-                    $query->whereIn('s.submission_id', function ($query) use ($word, $likePattern) {
-                        $query->select('p.submission_id')->from('publications AS p')
-                            ->join('publication_settings AS ps', 'p.publication_id', '=', 'ps.publication_id')
-                            ->where('ps.setting_name', '=', 'title')
-                            ->where(DB::raw('LOWER(ps.setting_value)'), 'LIKE', $likePattern)->addBinding($word);
-                    });
-                    $query->orWhereIn('s.submission_id', function ($query) use ($word, $likePattern) {
-                        $query->select('p.submission_id')->from('publications AS p')
-                            ->join('authors AS au', 'au.publication_id', '=', 'p.publication_id')
-                            ->join('author_settings AS aus', 'aus.author_id', '=', 'au.author_id')
-                            ->whereIn('aus.setting_name', [
-                                Identity::IDENTITY_SETTING_GIVENNAME,
-                                Identity::IDENTITY_SETTING_FAMILYNAME,
-                                'orcid'
-                            ])
-                            // Don't permit reviewers to search on author names
-                            ->when(is_array($this->assignedTo), function ($q) {
-                                $q->leftJoin('review_assignments AS ra', 'ra.submission_id', '=', 'p.submission_id')
-                                    ->whereIn('ra.reviewer_id', $this->assignedTo)
-                                    ->whereNull('ra.reviewer_id');
-                            })
-                            ->where(DB::raw('lower(aus.setting_value)'), 'LIKE', $likePattern)->addBinding($word);
-                    });
-                    if (ctype_digit((string) $word)) {
-                        $query->orWhere('s.submission_id', '=', $word);
-                    }
-                });
+            if(!empty($this->assignedTo)) {
+                // Holds a single random row to check whether we have any assignment
+                $q->leftJoinSub(fn (Builder $q) => $q
+                    ->from('review_assignments', 'ra')
+                    ->whereIn('ra.reviewer_id', $this->assignedTo)
+                    ->select(DB::raw('1 AS value'))
+                    ->limit(1),
+                    'any_assignment', 'any_assignment.value', '=', DB::raw('1')
+                );
             }
+            // Builds the filters
+            $q->where(fn (Builder $q) => $keywords
+                ->map(fn (string $keyword) => $q
+                    // Look for matches on the indexed data
```

Comment:
This filter can be slightly improved by nesting an extra check `Is submission published? Use indexed data : search elsewhere`.

That's why I asked if it was ok to let the [search indexer handle non-published content](https://github.com/pkp/pkp-lib/pull/8908#pullrequestreview-1381470606) :)